examples/demo-rollup/src/rollup.rs

```diff
@@ -67,16 +82,24 @@ pub async fn new_rollup_with_celestia_da(
     })
 }
 
-impl<Vm: Zkvm, DA: DaService<Error = anyhow::Error> + Clone> Rollup<Vm, DA> {
+impl<Vm: ProofSystem, DA: DaService<Error = anyhow::Error> + Clone> Rollup<Vm, DA> {
     /// Runs the rollup.
     pub async fn run(self) -> Result<(), anyhow::Error> {
         self.run_and_report_rpc_port(None).await
     }
-
     /// Runs the rollup. Reports rpc port to the caller using the provided channel.
     pub async fn run_and_report_rpc_port(
+        self,
+        channel: Option<oneshot::Sender<SocketAddr>>,
+    ) -> Result<(), anyhow::Error> {
+        self.run_with_prover_opt(channel, None).await
+    }
+
+    /// Runs the rollup. Reports rpc port to the caller using the provided channel.
+    pub async fn run_with_prover_opt(
```

Comment:
It looks like this method is never called with `prover: Some(_)` is it going to be used later?

Reply:
Yep, this is a placeholder for now just to keep the diff manageable/ The follow up will add integ tests using this method